libs/verifiable-credentials/src/main/java/uk/gov/di/ipv/core/library/verifiablecredential/validator/VerifiableCredentialValidator.java

```diff
@@ -221,46 +207,31 @@ private void validateClaimsSet(
         }
     }
 
-    private void validateCiCodes(VerifiableCredential vc) throws VerifiableCredentialException {
-        try {
-            var evidenceArray =
-                    OBJECT_MAPPER
-                            .valueToTree(vc.getClaimsSet().getClaim(VC_CLAIM))
-                            .path(VC_EVIDENCE);
-            if (evidenceArray.isArray()) {
-                List<CredentialEvidenceItem> credentialEvidenceList =
-                        OBJECT_MAPPER.treeToValue(
-                                evidenceArray, CREDENTIAL_EVIDENCE_ITEM_LIST_TYPE);
-                boolean anyUnrecognisedCiCodes = false;
-                for (CredentialEvidenceItem evidenceItem : credentialEvidenceList) {
-                    List<String> cis = evidenceItem.getCi();
-                    if (cis != null) {
-                        anyUnrecognisedCiCodes =
-                                cis.stream()
-                                        .anyMatch(
-                                                ciCode ->
-                                                        !configService
-                                                                .getContraIndicatorConfigMap()
-                                                                .containsKey(ciCode));
-                        if (anyUnrecognisedCiCodes) {
-                            break;
-                        }
-                    }
-                }
+    private void validateCiCodes(VerifiableCredential credential)
+            throws VerifiableCredentialException {
+        Stream<String> ciCodes;
+        if (credential.getCredential() instanceof IdentityCheckCredential icc) {
+            ciCodes =
+                    icc.getEvidence().stream()
+                            .map(IdentityCheck::getCi)
+                            .filter(Objects::nonNull)
+                            .flatMap(Collection::stream);
+        } else if (credential.getCredential() instanceof RiskAssessmentCredential rac) {
+            ciCodes =
+                    rac.getEvidence().stream()
+                            .map(RiskAssessment::getCi)
+                            .filter(Objects::nonNull)
+                            .flatMap(Collection::stream);
```

Comment:
A shame that we have to handle `IdentityCheckCredential` and `RiskAssessmentCredential` separately, but such is the Java life (wouldn't be a problem in TS 😄)